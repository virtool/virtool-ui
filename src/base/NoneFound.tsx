import React from "react";
import { cn } from "@/app/utils";
import Icon from "./Icon";
import { noneFoundStyle } from "./noneFoundStyle";

/**
 * A ListGroupItem component with a 'none found'-type message. Used in ListGroups when no data is available to populate
 * the list. For example, when no sample have been created.
 *
 * @param noun {string} the name of the items of which none were found (eg. samples)
 * @param noListGroup {boolean} don't include a ListGroup in the returned element
 */
interface NoneFoundProps {
    noun: string;
    className?: string;
}

export default function NoneFound({ noun, className }: NoneFoundProps) {
    return (
        <div className={cn(noneFoundStyle, className)}>
            <Icon name="info-circle" /> No {noun} found
        </div>
    );
}
