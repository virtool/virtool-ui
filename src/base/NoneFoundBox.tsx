import { cn } from "@/app/utils";
import { ReactNode } from "react";
import Box from "./Box";
import Icon from "./Icon";
import { noneFoundStyle } from "./noneFoundStyle";

interface NoneFoundBoxProps {
    children?: ReactNode;
    className?: string;
    noun: string;
}

export default function NoneFoundBox({
    className,
    noun,
    children,
}: NoneFoundBoxProps) {
    return (
        <Box className={cn(noneFoundStyle, "min-h-[30px]", className)}>
            <Icon name="info-circle" /> No {noun} found. &nbsp; {children}
        </Box>
    );
}
