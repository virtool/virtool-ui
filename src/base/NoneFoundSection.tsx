import React from "react";
import styled from "styled-components";
import BoxGroupSection from "./BoxGroupSection";
import Icon from "./Icon";
import { noneFoundStyle } from "./noneFoundStyle";

const StyledNoneFoundSection = styled(BoxGroupSection)`
    ${noneFoundStyle}
    justify-content: center;
`;

type NoneFoundSectionProps = {
    children?: React.ReactNode;
    noun: string;
};

export default function NoneFoundSection({
    children,
    noun,
}: NoneFoundSectionProps) {
    let childrenContainer;

    if (children) {
        childrenContainer = <span>. {children}.</span>;
    }

    return (
        <StyledNoneFoundSection>
            <Icon name="info-circle" /> No {noun} found{childrenContainer}
        </StyledNoneFoundSection>
    );
}
