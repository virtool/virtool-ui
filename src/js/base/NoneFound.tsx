import React from "react";
import styled from "styled-components";
import { Icon } from "./Icon";
import { noneFoundStyle } from "./noneFoundStyle";

/**
 * A ListGroupItem component with a 'none found'-type message. Used in ListGroups when no data is available to populate
 * the list. For example, when no sample have been created.
 *
 * @param noun {string} the name of the items of which none were found (eg. samples)
 * @param noListGroup {boolean} don't include a ListGroup in the returned element
 */
const StyledNoneFound = styled.div`
    ${noneFoundStyle}

    i.fas {
        margin-right: 5px;
    }
`;

interface NoneFoundProps {
    noun: string;
}

export const NoneFound = ({ noun }: NoneFoundProps) => {
    return (
        <StyledNoneFound>
            <Icon name="info-circle" /> No {noun} found
        </StyledNoneFound>
    );
};
