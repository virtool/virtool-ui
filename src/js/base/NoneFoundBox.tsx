import React from "react";
import styled from "styled-components";
import { Box } from "./Box";
import { Icon } from "./Icon";
import { noneFoundStyle } from "./noneFoundStyle";

interface NoneFoundBoxProps {
    children?: React.ReactNode;
    className?: string;
    noun: string;
}

const StyledNoneFoundBox = styled(Box)`
    ${noneFoundStyle}
    min-height: 30px;
`;

export function NoneFoundBox({ className, noun, children }: NoneFoundBoxProps) {
    return (
        <StyledNoneFoundBox as={Box} className={className}>
            <Icon name="info-circle" /> No {noun} found. &nbsp; {children}
        </StyledNoneFoundBox>
    );
}
