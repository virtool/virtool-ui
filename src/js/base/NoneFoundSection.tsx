import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import { BoxGroupSection } from "./BoxGroupSection";
import { Icon } from "./Icon";
import { noneFoundStyle } from "./noneFoundStyle";

const StyledNoneFoundSection = styled(BoxGroupSection)`
    ${noneFoundStyle}
    justify-content: center;
`;

export function NoneFoundSection({ children, noun }) {
    let childrenContainer;

    if (children) {
        childrenContainer = <span>. {children}.</span>;
    }

    return (
        <StyledNoneFoundSection>
            <Icon name="info-circle" /> No {noun} found{childrenContainer}
        </StyledNoneFoundSection>
    );
}

NoneFoundSection.propTypes = {
    children: PropTypes.node,
    noun: PropTypes.string.isRequired
};
