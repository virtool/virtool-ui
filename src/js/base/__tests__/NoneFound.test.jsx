import { shallow } from "enzyme";
import React from "react";
import { describe, expect, it } from "vitest";
import { NoneFound } from "../NoneFound";
import { NoneFoundBox } from "../NoneFoundBox";
import { NoneFoundSection } from "../NoneFoundSection";

describe("<NoneFound />", () => {
    it("should render with [noun='files'", () => {
        expect(shallow(<NoneFound noun="files" />)).toMatchSnapshot();
    });
});

describe("<NoneFoundBox />", () => {
    it("should render with [noun='boxes'", () => {
        expect(shallow(<NoneFoundBox noun="boxes" />)).toMatchSnapshot();
    });
});

describe("<NoneFoundSection />", () => {
    it("should render with [noun='sections'", () => {
        expect(shallow(<NoneFoundSection noun="sections" />)).toMatchSnapshot();
    });

    it("should render with [noun='sections'] and children", () => {
        expect(
            shallow(
                <NoneFoundSection noun="sections">
                    <span>foo</span>
                </NoneFoundSection>,
            ),
        ).toMatchSnapshot();
    });
});
