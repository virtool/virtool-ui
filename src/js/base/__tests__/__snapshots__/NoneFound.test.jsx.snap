// Vitest Snapshot v1

exports[`<NoneFound /> > should render with [noun='files' 1`] = `
<styled.div>
  <Icon
    name="info-circle"
  />
   No 
  files
   found
</styled.div>
`;

exports[`<NoneFoundBox /> > should render with [noun='boxes' 1`] = `
<Styled(Box)
  as={
    {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": [],
      "componentStyle": e {
        "baseHash": 816878857,
        "baseStyle": undefined,
        "componentId": "sc-jSUZER",
        "isStatic": false,
        "rules": [
          "
    border: ",
          [Function],
          ";
    border-radius: ",
          [Function],
          ";
    box-sizing: border-box;
    cursor: ",
          [Function],
          ";
    margin-bottom: 15px;
    padding: 10px 15px;
    position: relative;

    &:hover {
        ",
          [Function],
          "
    }
",
        ],
        "staticRulesId": "",
      },
      "foldedComponentIds": [],
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "sc-jSUZER",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
>
  <Icon
    name="info-circle"
  />
   No 
  boxes
   found. Â  
</Styled(Box)>
`;

exports[`<NoneFoundSection /> > should render with [noun='sections' 1`] = `
<Styled(BoxGroupSection)>
  <Icon
    name="info-circle"
  />
   No 
  sections
   found
</Styled(BoxGroupSection)>
`;

exports[`<NoneFoundSection /> > should render with [noun='sections'] and children 1`] = `
<Styled(BoxGroupSection)>
  <Icon
    name="info-circle"
  />
   No 
  sections
   found
  <span>
    . 
    <span>
      foo
    </span>
    .
  </span>
</Styled(BoxGroupSection)>
`;
